js/checkout.js

```diff
@@ -180,6 +200,70 @@ class Blockonomics {
     redirect_to_finish_order() {
         window.location.href = this.data.finish_order_url
     }
+
+    _set_refresh_loading(loading=false) {
+        if(loading) {
+            this._refresh.querySelector('.blockonomics-icon-refresh').classList.add('spin')
+            this._refresh.setAttribute('disabled', 'disabled')
+        } else {
+            this._refresh.querySelector('.blockonomics-icon-refresh').classList.remove('spin')
+            this._refresh.removeAttribute('disabled')
+        }
+    }
+
+    refresh_order() {
+        let url = `${this.data.api_url}?get_order=${this.get_url_param('show_order')}&crypto=${this.get_url_param('crypto')}`
+
+        this._set_refresh_loading(true)
+
+        // Stop Progress Counter
+        clearInterval(this.progress.interval)
+        
+        fetch(url, {method: 'GET'}).then(
+            res => {
+                this._set_refresh_loading(false)
+                if (res.status == 200) {
+                    res.json().then(data => {
+                        this._update_order_params(data)
+                    })
+                }
+            },
+            err => {
+                // Blocked by Network Errors such as CORS, Offline, Server blocking request, etc
+                this._set_refresh_loading(false)
+                this._fallback_refresh_order()
+            }
+        )
+    }
+
+    _update_order_params(data) {
+        // Updates the Dynamic Parts of Page
+        
+        const crypto_amount = this._satoshi_to_amount(parseFloat(data.satoshi))
+        const fiat_conversion_rate = this._price_per_crypto(crypto_amount, parseFloat(data.value))
+
+        this._amount_input.value = crypto_amount
+        this._crypto_rate.innerHTML = fiat_conversion_rate
+
+        this.reset_progress()
+    }
+
+    _fallback_refresh_order() {
+        location.reload()
+    }
+
+    _satoshi_to_amount(satoshi) {
+        return satoshi/1.0e8
+    }
+
+    _price_per_crypto(crypto, fiat_amount) {
+        return Number(fiat_amount/crypto).toFixed(2) // Show Fixed Decimal Points rather than rounding off
+    }
```

Comment:
One line functions are a bad practice particularly particularly when the function itself is called from only one place
https://mytwoocents.com/2017/03/25/one-line-methods-can-easily-become-an-antipattern/
In C we have macros for the this same reason

Reply:
It makes sense, I've updated the PR for the same. I haven't yet replaced `redirect_to_finish_order` method as it was not part of this PR.